doc/development_process/index.rst

```diff
@@ -0,0 +1,310 @@
+the code base.
+
+The Zephyr project uses GitHub for code reviews and GIT tree management. When
+submitting a change or an enhancement to any Zephyr component, a developer should
+use GitHub. GitHub automatically assigns a responsible reviewer on a component
+basis, as defined in the CODEOWNERS file stored with the code tree in the Zephyr
+project repository. A limited set of release managers are allowed to merge a PR
+into the master branch once reviews are complete.
+
+Continuous Integration
+=======================
+
+All changes submitted to GitHub are subject to sanity tests that are run on
+emulated platforms and architectures to identify breakage and regressions that
+can be immediately identified.  Sanity testing additionally performs build tests
+of a representative number of boards and platforms (subject to hardware
+availability in the CI infrastructure). Documentation changes are also verified
+through review and build testing to verify doc generation will be successful.
+
+Any failures found during the CI test run will result in a negative review.
+Developers are expected to fix issues and rework their patches and submit again.
+
+The CI infrastructure currently runs the following tests:
+
+- Run '''checkpatch''' for code style issues (can vote -1 on errors)
+- Gitlint: Git commit style based on project requorements
+- License Check: Check for conflicting licenses
+- Run '''sanitycheck''' script:
+
+  - Run kernel tests in QEMU (can vote -1 on errors)
+  - Build various samples for different boards (can vote -1 on errors)
+
+
+Dealing with Proposals and RFCs
+*******************************
+
+Feature or enhancement proposals for Zephyr are used to propose a new design or
+modify an existing feature and get it approved. Currently, the developer mailing
+list is used to make such proposals. To better integrate with the current tools
+used in the project and to enable traceability and archiving of such proposals
+the following is required:
+
+- A well-defined and unified format for enhancement proposals
+- Central repository for maintaining such proposals
+- A well-defined process for reviews and approval
+
+Procedures
+==========
+
+Maintain such proposals in a document format (similar to the documentation
+format used in the kernel source tree, using restructured Text) in a GitHub
+repository. The review and approval process of GitHub could be used for
+approving such proposals as well, so there will be no need for a dedicated tool.
+
+Process
+========
+
+- Open an Improvement GitHub Issue with an abstract and the proposal itself
+- Send an RFC with the abstract to the mailing list with an abstract and the
+  link to the detailed proposal
+- Approve/Reject/Defer based on feedback and comments
+
+Bug Reporting and Feature Tracking
+***********************************
+
+To maintain traceability and relation between proposals, changes, features, and
+issues, cross-referencing a commit with a GitHub Issue and vice versa should be
```

Comment:
Sidetrack: Shouldn't (we) reviewers actually enforce this?

Reply:
we should I think.